src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1806,9 +1808,11 @@ export const handleUpdateLogic = [
           .items(
             Joi.object({
               field: Joi.string().required(),
-              state: Joi.string().required(),
-              value: Joi.string().required(),
-              ifValueType: Joi.string(),
+              state: Joi.string()
+                .valid(...Object.values(LogicConditionState))
+                .required(),
+              value: Joi.any().required(),
```

Comment:
can I check why this is `any`? would it be possible to experiment with the various logic field types to determine exactly what type is being returned here? imo we should always know the exact shape of our APIs

Reply:
Thanks! I typed this based on the schema in `form_logic.server.schema.ts`. From what I can see though, it's either a number or string, so I've tightened accordingly. I also left open the possibility for it to be a number array or string array because we allow for those types in `src/types/form_logic.ts`

```
export interface ICondition {
  field: IFieldSchema['_id']
  state: LogicConditionState
  value: string | number | string[] | number[]
  ifValueType?: LogicIfValue
}
```